File: pykde-bugs/gaccel.cc
#include <iostream>

#include <kapplication.h>
#include <kcmdlineargs.h>
#include <kmainwindow.h>
#include <kglobalaccel.h>
#include <kshortcut.h>

class MainWindow : public KMainWindow
{
    Q_OBJECT;

    public:
      MainWindow();
      ~MainWindow();

    private:
    	KGlobalAccel *gaccel;

    public slots:
    	void slot_action();
};

MainWindow::MainWindow() : KMainWindow()
{
      gaccel = new KGlobalAccel(this);
      gaccel->insert("action", "Action", "", KShortcut("Ctrl+Alt+u"), 0, this, SLOT(slot_action()));
      gaccel->updateConnections();
}

MainWindow::~MainWindow()
{
    delete gaccel;
}

void
MainWindow::slot_action()
{
      std::cout << "Inside slot_action()" << std::endl;
}

int
main (int argc, char **argv)
{
    KCmdLineArgs::init(argc, argv, "test", "test", "test", "1.0");
    KApplication app;
    MainWindow *window = new MainWindow;

    app.setTopWidget(window);
    window->show();
    return app.exec();
}

#include "gaccel_moc.cc"

File: pykde-bugs/gaccel.py
#! /usr/bin/env python

"""http://www.riverbankcomputing.com/pipermail/pyqt/2007-August/016865.html"""

import sys

import qt
import kdeui
import kdecore

def main():
    application = kdecore.KApplication(sys.argv, 'test app')
    main_window = MainWindow(None, 'main window')
    main_window.show()
    application.exec_loop()

class MainWindow(kdeui.KMainWindow):
    def __init__ (self, *args):
        kdeui.KMainWindow.__init__(self, *args)

        self.global_accel = kdecore.KGlobalAccel(self)
        self.global_accel.insert('action', 'Action', '', kdecore.KShortcut('Ctrl+Alt+U'),
                kdecore.KShortcut(), self.slot_action)

        self.global_accel.updateConnections()

    def slot_action(self):
        print "Inside slot_action()"

if __name__ == '__main__':
    main()
